### ISSUE TITLE
Confusing "false positive" reportUnnecessaryComparison with new `Type[Any]` behaviour

### ISSUE THREAD

**Issue by RobertCraigie** [microsoft/pyright#2684]
**Describe the bug**

```py
from typing import Type, Any

class Generator:
    foo: int

def foo(base: Type[Any]) -> None:
    if base == Generator:
        print('Given type is Generator')
    else:
        print('Unexpected type')

foo(Generator)
```

Pyright reports:

```
bar.py:9:8 - error: Condition will always evaluate to False since the types "type" and "Type[Generator]" have no overlap (reportUnnecessaryComparison)
```

This diagnostic can be fixed by changing `Type[Any]` to `Type[object]`, however this is not an obvious solution.

```py
def foo(base: Type[object]) -> None:
    if base == Generator:
        print('Given type is Generator')
    else:
        print('Unexpected type')
```

**Expected behavior**

I do not know if this behaviour is expected or not but either the error message should be improved to suggest using `Type[object]` or it should not error.

**VS Code extension or command-line**

Command-line: 1.194

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.195, which I just published. It will also be included in the next release of pylance.
